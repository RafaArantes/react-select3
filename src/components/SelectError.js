import React from 'react'
import PropTypes from 'prop-types'


const SelectError = ({ error }, { cssClassNamePrefix }) => {
  return (typeof error === 'string') ?
    (<span className={`${cssClassNamePrefix}__error`}>{ error }</span>)
    : null;
};

SelectError.contextTypes = {
  cssClassNamePrefix: PropTypes.string,
};

SelectError.propTypes = {
  error: PropTypes.oneOfType([
    PropTypes.bool,
    PropTypes.string,
  ]),
};

export default SelectError
