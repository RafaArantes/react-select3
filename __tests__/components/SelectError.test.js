import SelectError from '../../src/components/SelectError'


describe('SelectError', () => {
  it('Should return empty string if error prop weren\'t passed render', () => {
    const component = shallow(<SelectError />, selectComponentContext);

    expect(component).toMatchSnapshot()
  });

  it('Should show error message when provided', () => {
    const error = 'Test error';
    const component = shallow(<SelectError error={ error }/>, selectComponentContext);

    expect(component).toMatchSnapshot();
    expect(component.text()).toBe(error)
  })
});
