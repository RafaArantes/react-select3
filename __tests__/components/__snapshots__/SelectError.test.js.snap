// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectError Should return empty string if error prop weren't passed render 1`] = `""`;

exports[`SelectError Should show error message when provided 1`] = `
<span
  className="rs3__error"
>
  Test error
</span>
`;
